sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/IcebergWriteSchemaTransformProvider.java

```diff
@@ -160,7 +176,14 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
 
       IcebergIO.WriteRows writeTransform =
           IcebergIO.writeRows(configuration.getIcebergCatalog())
-              .to(TableIdentifier.parse(configuration.getTable()));
+              .to(
+                  new PortableIcebergDestinations(
+                      configuration.getTable(),
```

Comment:
We should also update the documentation of table (string) to clarify how one can use a formatted string to specify a dynamic destination table.

Reply:
We don't have any class documentation in `IcebergIO.java` yet. Can we add some in a future PR?

We can update the dataflow documentation for now though: https://cloud.google.com/dataflow/docs/guides/write-to-iceberg#configuration

Reply:
Yeah, adding in a future PR is fine.

Reply:
Adding documentation in #32621